cloud/ec2.go

```diff
@@ -1077,6 +1077,17 @@ func (m *ec2Manager) AttachVolume(ctx context.Context, h *host.Host, attachment
 		return errors.Wrap(err, "error creating client")
 	}
 
+	if attachment.DeviceName == "" {
+		deviceName := generateDeviceNameForVolume()
+		exists, err := host.HostExistsWithVolumeWithDeviceName(deviceName)
+		if err != nil {
+			return errors.Wrapf(err, "error querying for volume with device name")
+		}
```

Comment:
shouldn't we retry?

Reply:
I don't want to be stuck in an infinite loop or something in case there's some ongoing database issue, but I suppose we can rely on the context to timeout in that case.